### PR TITLE
build and test properly in gitlab

### PR DESCRIPTION
**Pull request by alejandroEsc**
**What this PR does / why we need it**:
added build directory and files to get gitlab to build and test properly.

**Release note**:

```NONE
```

### CODE REVIEW

**Comment by alejandroEsc** [comment]
DO NO merge yet, adding a few things.

**Comment by notjames** [comment]
Can't merge until build succeeds in pipelines.

File: vault-etcd/tls-generator/generate-certs.sh [hunk 1]
```diff
@@ -49,7 +49,8 @@ HOSTNAME_PREFIX="etcd-vault-etcd"
 NAMESPACE="vault2"
```
Comment (notjames) [inline vault-etcd/tls-generator/generate-certs.sh]: vault2?

**Comment by alejandroEsc** [comment]
@notjames good catches, revised approach, should be better now.

**Pull request merged**